elliptic-curve/src/arithmetic.rs

```diff
@@ -84,3 +84,21 @@ pub trait PrimeCurveArithmetic:
     /// Prime order elliptic curve group.
     type CurveGroup: group::prime::PrimeCurve<Affine = <Self as CurveArithmetic>::AffinePoint>;
 }
+
+/// Perform a batched conversion to affine representation on a sequence of projective points
+/// at an amortized cost that should be practically as efficient as a single conversion.
+/// Internally, implementors should rely upon `InvertBatch`.
+pub trait ToAffineBatch: CurveArithmetic {
+    /// Converts a batch of points in their projective representation into the affine ones.
+    /// /// This variation takes a const-generic array and thus does not require `alloc`.
+    fn to_affine_batch_array<const N: usize>(
+        points: &[Self::ProjectivePoint; N],
+    ) -> [Self::AffinePoint; N];
+
+    /// Converts a batch of points in their projective representation into the affine ones.
+    /// However, this also requires to make dynamic allocations and as such requires `alloc`.
+    #[cfg(feature = "alloc")]
+    fn to_affine_batch_slice<B: FromIterator<Self::AffinePoint>>(
```

Comment:
I think `group` already has a method for this: [`group::Curve::batch_normalize`](https://docs.rs/group/latest/group/trait.Curve.html#method.batch_normalize)

A trait for `to_affine_batch_array` still seems interesting, but should probably be similarly defined in terms of `group::Curve` so you can impl it for `ProjectivePoint`.

Reply:
I'm ok with this, but then better change `batch_invert_slice` back to `batch_invert` to keep the same line

Reply:
I also renamed to `Normalize` and added a `to_affine()` method, so we'll really be in line with `Invert`